src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -240,44 +198,44 @@ internal static object DefaultToType(IConvertible value, Type targetType, IForma
                 return value;
             }
 
-            if (ReferenceEquals(targetType, ConvertTypes[(int)TypeCode.Boolean]))
+            if (ReferenceEquals(targetType, typeof(bool)))
```

Comment:
Does the new optimization work with ReferenceEquals or should this be changed to `targetType == typeof(bool)`?

Reply:
It works. `ReferenceEquals` is a C# compiler intrinsic and maps to a reference comparison. The intrinsic expansion operates on simplifying the reference comparison.

`==` operator on `System.Type` is turned into a reference comparison by RyuJIT when one side of the comparison is known to be a runtime type (result of `typeof` or `object.GetType`).

Reply:
But it would look better if this is written as `targetType == typeof(bool)`.